databricks/koalas/indexes.py

```diff
@@ -2077,6 +2077,59 @@ def holds_integer(self):
         """
         return isinstance(self.spark.data_type, IntegralType)
 
+    def intersection(self, other):
```

Comment:
Shall we add the return type `-> "Index"`? Thanks!

Reply:
Thanks, @xinrong-databricks !